File: packages/java/typescript-generator/src/test/java/com/vaadin/hilla/parser/plugins/backbone/simpletype/SimpleTypeEndpoint.java
/*
 * Copyright 2000-2025 Vaadin Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */
package com.vaadin.hilla.parser.plugins.backbone.simpletype;

import com.vaadin.hilla.parser.testutils.annotations.Endpoint;
import java.math.BigDecimal;
import java.math.BigInteger;

@Endpoint
public class SimpleTypeEndpoint {
    public void doSomething() {
        // no-op
    }

    public int[] getArray() {
        return new int[] { 1, 2, 3 };
    }

    public BigDecimal getBigDecimal() {
        return BigDecimal.valueOf(0);
    }

    public BigInteger getBigInteger() {
        return BigInteger.valueOf(0);
    }

    public boolean getBoolean() {
        return true;
    }

    public Boolean getBooleanWrapper() {
        return getBoolean();
    }

    public byte getByte() {
        return 0;
    }

    public Byte getByteWrapper() {
        return getByte();
    }

    public char getChar() {
        return 'a';
    }

    public Character getCharWrapper() {
        return getChar();
    }

    public double getDouble() {
        return 0.0;
    }

    public Double getDoubleWrapper() {
        return getDouble();
    }

    public float getFloat() {
        return (float) 0.0;
    }

    public Float getFloatWrapper() {
        return getFloat();
    }

    public int getInteger() {
        return 0;
    }

    public Integer getIntegerWrapper() {
        return getInteger();
    }

    public long getLong() {
        return 0;
    }

    public Long getLongWrapper() {
        return getLong();
    }

    public short getShort() {
        return 0;
    }

    public Short getShortWrapper() {
        return getShort();
    }

    public String getString() {
        return "test";
    };

    protected String getProtectedValue() {
        return "protected";
    }

    protected void setProtectedValue(String protectedValue) {
        // ignore
    }
}

File: packages/java/typescript-generator/src/test/resources/com/vaadin/hilla/parser/plugins/backbone/simpletype/snapshots/SimpleTypeEndpoint.ts
import type { EndpointRequestInit } from '@vaadin/hilla-frontend';
import client from './connect-client.default.js';

export async function doSomething(init?: EndpointRequestInit): Promise<void> {
  return client.call('SimpleTypeEndpoint', 'doSomething', {}, init);
}

export async function getArray(init?: EndpointRequestInit): Promise<Array<number> | undefined> {
  return client.call('SimpleTypeEndpoint', 'getArray', {}, init);
}

export async function getBigDecimal(init?: EndpointRequestInit): Promise<number | undefined> {
  return client.call('SimpleTypeEndpoint', 'getBigDecimal', {}, init);
}

export async function getBigInteger(init?: EndpointRequestInit): Promise<number | undefined> {
  return client.call('SimpleTypeEndpoint', 'getBigInteger', {}, init);
}

export async function getBoolean(init?: EndpointRequestInit): Promise<boolean> {
  return client.call('SimpleTypeEndpoint', 'getBoolean', {}, init);
}

export async function getBooleanWrapper(init?: EndpointRequestInit): Promise<boolean | undefined> {
  return client.call('SimpleTypeEndpoint', 'getBooleanWrapper', {}, init);
}

export async function getByte(init?: EndpointRequestInit): Promise<number> {
  return client.call('SimpleTypeEndpoint', 'getByte', {}, init);
}

export async function getByteWrapper(init?: EndpointRequestInit): Promise<number | undefined> {
  return client.call('SimpleTypeEndpoint', 'getByteWrapper', {}, init);
}

export async function getChar(init?: EndpointRequestInit): Promise<string> {
  return client.call('SimpleTypeEndpoint', 'getChar', {}, init);
}

export async function getCharWrapper(init?: EndpointRequestInit): Promise<string | undefined> {
  return client.call('SimpleTypeEndpoint', 'getCharWrapper', {}, init);
}

export async function getDouble(init?: EndpointRequestInit): Promise<number> {
  return client.call('SimpleTypeEndpoint', 'getDouble', {}, init);
}

export async function getDoubleWrapper(init?: EndpointRequestInit): Promise<number | undefined> {
  return client.call('SimpleTypeEndpoint', 'getDoubleWrapper', {}, init);
}

export async function getFloat(init?: EndpointRequestInit): Promise<number> {
  return client.call('SimpleTypeEndpoint', 'getFloat', {}, init);
}

export async function getFloatWrapper(init?: EndpointRequestInit): Promise<number | undefined> {
  return client.call('SimpleTypeEndpoint', 'getFloatWrapper', {}, init);
}

export async function getInteger(init?: EndpointRequestInit): Promise<number> {
  return client.call('SimpleTypeEndpoint', 'getInteger', {}, init);
}

export async function getIntegerWrapper(init?: EndpointRequestInit): Promise<number | undefined> {
  return client.call('SimpleTypeEndpoint', 'getIntegerWrapper', {}, init);
}

export async function getLong(init?: EndpointRequestInit): Promise<number> {
  return client.call('SimpleTypeEndpoint', 'getLong', {}, init);
}

export async function getLongWrapper(init?: EndpointRequestInit): Promise<number | undefined> {
  return client.call('SimpleTypeEndpoint', 'getLongWrapper', {}, init);
}

export async function getShort(init?: EndpointRequestInit): Promise<number> {
  return client.call('SimpleTypeEndpoint', 'getShort', {}, init);
}

export async function getShortWrapper(init?: EndpointRequestInit): Promise<number | undefined> {
  return client.call('SimpleTypeEndpoint', 'getShortWrapper', {}, init);
}

export async function getString(init?: EndpointRequestInit): Promise<string | undefined> {
  return client.call('SimpleTypeEndpoint', 'getString', {}, init);
}
